railgun/internal/ctrlutils/ingress-status.go

```diff
@@ -150,29 +157,45 @@ func UpdateIngressV1(ctx context.Context, logger logr.Logger, svc file.FService,
 				continue
 			}
 
-			var status []apiv1.LoadBalancerIngress
-			sort.SliceStable(status, lessLoadBalancerIngress(status))
-			curIPs := curIng.Status.LoadBalancer.Ingress
-
-			status = SliceToStatus(ips)
-			if ingressSliceEqual(status, curIPs) {
-				log.Debugf("no change in status, update ingress v1 skipped")
-				return nil
+			existingHash, err := util.GetValue(ingresKey)
+			if err != nil {
+				log.Infof("v1ingress %s not processed yet. ", ingresKey)
+			} else {
+				curHash, err := hashstructure.Hash(*curIng, hashstructure.FormatV2, nil)
+				if err != nil {
+					panic(err)
+				}
+				if existingHash == curHash {
+					log.Info("no change in v1 ingress %s, skip updating.", ingresKey)
```

Comment:
Keep [K8S verbosity level recommendations](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md) in mind when adding logs, especially for logs within loops over all Ingresses or similar.

Log entries like "status unchanged" are not usually of much interest, and should only appear on `V(4)` or `V(5)`.

Log entries for steps leading up to some event (e.g. the "not processed yet" log above) can appear on `V(3)`.

Lower levels (`V(2)` or `V(1)`) should be reserved for the conclusion of a process (e.g. the "successfully updated" below) and include information that describes the whole even (e.g. rather than simply "successfully updated", provide the new and previous IP, to indicate the new and previous state).